### PR TITLE
Add unit tests for a variety of chart aspects

### PR DESCRIPTION
**Pull request by cognifloyd**
Part of #28

- correct helm unittest usage with more than one test file
- Add unit tests for
    - required Labels
    - SecurityContext
    - imagePullSecrets and PullPolicy
    - Resources
    - Pod placement (NodeSelector, Tolerations, Affinity)
    - dnsConfig, dnsPolicy
    - ServiceAccount
    - postStartScript

### CODE REVIEW

**Comment by cognifloyd** [comment]
@armab I added a bunch of tests here. I'll stop adding more to this PR and wait for your review.

File: tests/unit/labels_test.yaml
Comment (cognifloyd) [inline tests/unit/labels_test.yaml:200]: Maybe fix this in a follow up

File: tests/unit/post_start_script_test.yaml
Comment (cognifloyd) [inline tests/unit/post_start_script_test.yaml:109]: Fix this in a follow up (need to drop the condition since they always have a script)

File: tests/unit/resources_test.yaml
Comment (cognifloyd) [inline tests/unit/resources_test.yaml:96]: Maybe we should make this configurable in a follow up.

File: tests/unit/service_account_test.yaml
Comment (cognifloyd) [inline tests/unit/service_account_test.yaml:96]: That's odd. Maybe add `st2client.serviceAccount.attach` in a follow up.

**Review by cognifloyd — COMMENTED** [review]
The tests reveal a few inconsistencies that we might want to fix in follow-up PRs.

**Review by armab — APPROVED** [review]
Thanks @cognifloyd :+1: 

I'd also say add a few quick unit tests for the https://github.com/StackStorm/stackstorm-ha/pull/218 and let's merge it to call that done.

**Comment by cognifloyd** [comment]
@armab OK. I added tests for `stackstorm/sensor-mode=all-sensors-in-one-pod` and `stackstorm/sensor-mode=one-sensor-per-pod` in https://github.com/StackStorm/stackstorm-ha/pull/288/commits/3c86fd6413ac94cb33c74a8bec7e4c9058234943. I can add the tests for just the sensor hash-partitioning feature in #218.

Please check out https://github.com/StackStorm/stackstorm-ha/pull/288/commits/3c86fd6413ac94cb33c74a8bec7e4c9058234943, and then merge this if you are happy with those tests.

**Comment by armab** [comment]
@cognifloyd It all looks already great. Feel free to merge!

I really meant it would be nice to add tests in the https://github.com/StackStorm/stackstorm-ha/pull/218 so we can merge that one too, but with `st2sensors_test.yaml` added here, the coverage will be even better :+1:

**Pull request merged**
